detection_rules/config.py

```diff
@@ -197,6 +198,10 @@ def __post_init__(self):
         if 'name' not in self.packages['package']:
             raise ValueError('Missing the `name` field defined in packages.yaml.')
 
+        if 'registry_data' not in self.packages['package'] and not self.bypass_version_lock:
```

Comment:
registry data is only required for building fleet packages. What is the use case here?

Reply:
We're adding more verbose logging for when registry_data isn't available. We could do this in other places, but this is a centralized / already loaded opportunity to catch this missing field.

Reply:
based on the discussion. will punt to another issue to increase the scope of registry support.